test/RecipeBot.Persistence.Test/RecipeDataEntryCollectionRepositoryTest.cs

```diff
@@ -0,0 +1,312 @@
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Fish)]
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Meat)]
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Drinks)]
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Pastry)]
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Dessert)]
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Snack)]
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Vegetarian)]
+    [MemberData(nameof(GetCategoryTests), RecipeCategory.Vegan)]
+    public async Task Given_seeded_database_when_loading_entries_with_category_and_match_found_returns_filtered_data_ordered_by_id(
+        IReadOnlyCollection<RecipeEntity> recipeEntries, RecipeCategory categoryToFilter, IEnumerable<RecipeEntity> expectedRecipes)
+    {
+        // Setup
+        using (RecipeBotDbContext context = CreateContext())
+        {
+            await context.Database.EnsureCreatedAsync();
+            await context.RecipeEntities.AddRangeAsync(recipeEntries);
+            await context.SaveChangesAsync();
+
+            context.ChangeTracker.Clear();
+
+            var repository = new RecipeDataEntryCollectionRepository(context);
+
+            // Call
+            IReadOnlyList<RecipeEntryData> entries = await repository.LoadRecipeEntriesAsync(categoryToFilter);
+
+            // Assert
+            entries.Should().BeInAscendingOrder(s => s.Id).And.BeEquivalentTo(
+                expectedRecipes,
+                options => options.ExcludingMissingMembers()
+                                  .WithMapping<RecipeEntryData>(e => e.RecipeEntityId, s => s.Id)
+                                  .WithMapping<RecipeEntryData>(e => e.RecipeTitle, s => s.Title)
+                                  .WithMapping<AuthorEntity, RecipeEntryData>(e => e.AuthorName, s => s.AuthorName));
+        }
+    }
+
+    public void Dispose()
+    {
+        GC.SuppressFinalize(this);
+        connection.Dispose();
+    }
+
+    private static IReadOnlyCollection<RecipeEntity> GetDatabaseSeed(PersistentRecipeCategory category)
+    {
+        var fixture = new Fixture();
+        var authorOne = new AuthorEntity
+        {
+            AuthorName = fixture.Create<string>(),
+            AuthorImageUrl = fixture.Create<string>()
+        };
+        var authorTwo = new AuthorEntity
+        {
+            AuthorName = fixture.Create<string>(),
+            AuthorImageUrl = fixture.Create<string>()
+        };
+
+        var recipeOne = new RecipeEntity
+        {
+            RecipeEntityId = fixture.Create<int>(),
+            Author = authorOne,
+            RecipeCategory = category,
+            RecipeTitle = fixture.Create<string>()
+        };
+        var recipeTwo = new RecipeEntity
+        {
+            RecipeEntityId = fixture.Create<int>(),
+            Author = authorTwo,
+            RecipeCategory = category == PersistentRecipeCategory.Other ? PersistentRecipeCategory.Dessert : category,
```

Comment:
The false clause should  be something else to make the test correct. Otherwise all other cases will return a collection of 3 recipes (and no filtering took place)